enhancements/cluster-logging/forwarding-json-structured-logs.md

```diff
@@ -0,0 +1,354 @@
+---
+title: forwarding-json-structured-logs
+
+authors:
+  - "@alanconway"
+
+reviewers:
+  - "@ewolinetz"
+  - "@lukas-vlcek"
+  - "@jcantril"
+  - "@sichvoge"
+
+approvers:
+  - "@jcantril"
+
+creation-date: 2020-10-27
+
+status: provisional
+
+see-also:
+  - [Elasticsearch Pipeline Processing](cluster-logging-es-pipeline-processing.md)
+  - [Forwarder Label Selector](forwarder-label-selector.md)
+---
+
+# Forwarding JSON Structured Logs
+
+## Release Sign-off Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+When applications write structured JSON logs, consumers want to access fields of JSON log entries for indexing and other purposes.
+The current logging [data model][data_model] stores the log entry as a JSON *string*, not a JSON *object*.
+Consumers can't access the log entry fields without a second JSON parse of this string.
+
+The current implementation also 'flattens' labels to work around Elasticsearch limitations.
+
+To illustrate, given a log entry `{"name":"fred","home":"bedrock"}` from a container with the label `app.kubernetes.io/name="flintstones"`. The current output record is:
+
+```json
+{
+  "message":"{\"name\":\"fred\",\"home\":\"bedrock\"}",
+  "kubernetes":{"labels":{"app_kubernetes_io/name": "flintstones", ...}},
```

Comment:
nit but we this doesnt demonstrate the final output with "flat_labels"

Reply:
Good catch, will fix.